src/DotLiquid/StandardFilters.cs

```diff
@@ -931,6 +931,53 @@ public static IEnumerable Compact(object input)
             ary.RemoveAll(item => item == null);
             return ary;
         }
+
+        /// <summary>
+        /// Creates an array including only the objects with a given property value, or any truthy value by default.
+        /// </summary>
+        /// <param name="input">an array to be filtered</param>
+        /// <param name="property">The name of the property to filter by</param>
+        /// <param name="target_value">Value to retain, if null object containing this property are retained</param>
+        /// <returns></returns>
+        public static IEnumerable Where(object input, string property, string target_value = null)
+        {
+            if (input == null)
+                return null;
+
+            if (property.IsNullOrWhiteSpace())
+                throw new ArgumentNullException(paramName: nameof(property), message: $"'{nameof(property)}' cannot be null or empty.");
+
+            List<object> inputList = (input is IEnumerable enumerableInput) ? enumerableInput.Cast<object>().ToList() : null;

```

Comment:
If this line returns null the next line will likely fail with null deref exception. Please add test coverage for this and fix as appropiate.

Reply:
Change type of input to IEnumerable, removed superfluous if/else and added test case coverage for nulls